### ISSUE TITLE
not getting any dice score for enhancing tumor and tumor core where as getting good dice score for whole tumor

### ISSUE THREAD

**Issue by sneh-debug** [ellisdg/3DUnetCNN#256]
@ellisdg I trained original unet and after 87 epochs training stopped due to early stopping and when i check dice score for validation images i got dice score for whole tumor whereas 0 dice score for the other two. am i missing something?

**Comment by ellisdg** [comment]
If I remember correctly, last time I tried the original UNet I got similar results. Try using the weighted dice coefficient loss function instead of the normal one. This will place equal weight to each label in the label map. With the normal dice, it weights each voxel equally, so the labels with the most voxels get weighted much heavier than the labels with fewer voxels, such as ET. You can also try turning permutations off, because I think the BraTS dataset is big enough that the original UNet model (hopefully) won't overfit too much.
